src/test/test_controller.c

```diff
@@ -1689,6 +1691,187 @@ test_download_status_bridge(void *arg)
+
+int
+mock_we_want_to_fetch_flavor(const or_options_t *options,
+                             int flavor) {
+  if (options->FetchUselessDescriptors ||
+      flavor == FLAV_NS || flavor == FLAV_MICRODESC) {
+    return 1;
+  }
+  return 0;
+}
+
+/**  Mock the function that retrieves consensus from
+ * cache. These use a global variable so that they can
+ * be cleared from within the test. The actual code
+ * retains the pointer to the consensus data, but
+ * we are doing this here, to prevent memory leaks
+ * from within the tests */
+static cached_dir_t *
+mock_dirserv_get_consensus(const char *flavor_name);
+
+static cached_dir_t *
+mock_dirserv_get_consensus(const char *flavor_name)
+{
+  if (!strcmp(flavor_name, "ns")) {
+    mock_ns_consensus_cache = tor_malloc_zero(sizeof(cached_dir_t));
+    mock_ns_consensus_cache->dir = tor_strdup("mock_ns_consensus");
+    return mock_ns_consensus_cache;
+  } else {
+    mock_microdesc_consensus_cache = tor_malloc_zero(sizeof(cached_dir_t));
+    mock_microdesc_consensus_cache->dir = tor_strdup(
+                                            "mock_microdesc_consensus");
+    return mock_microdesc_consensus_cache;
+  }
+}
+
+/** Mock for the function that returns the name of the
+ * file containing the consensuses. */
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus);
+
+char *
+mock_networkstatus_get_cache_fname(int flav,
+                                   const char *flavor_name,
+                                   int unverified_consensus)
+{
+  /** A function needs to do something with
+   * all variables that are passed as arguments */
+  flav = flav & 1;
+  unverified_consensus = unverified_consensus & 1;
+  return tor_strdup(flavor_name);
+}
+
+/** Mock the function that retrieves consensuses
+ *  from a files in the directory. */
+tor_mmap_t *
+mock_tor_mmap_file(const char* filename);
+
+tor_mmap_t *
+mock_tor_mmap_file(const char* filename)
+{
+  tor_mmap_t *res;
+  res = tor_malloc_zero(sizeof(tor_mmap_t));
+  if (!strcmp(filename, "ns")) {
+    res->data = "mock_ns_consensus";
+  }
+  if (!strcmp(filename, "microdesc")) {
+    res->data = "mock_microdesc_consensus";
+  }
+  res->size = strlen(res->data);
+  return res;
+}
+
+int
+mock_tor_munmap_file(tor_mmap_t *handle);
+
+/** Mock the function that clears file data
+ * loaded into the memory */
+int
+mock_tor_munmap_file(tor_mmap_t *handle)
+{
+  tor_free(handle);
+  return 0;
+}
+
+static void
+test_getinfo_helper_dir_from_file(void *arg)
+{
+  /* We just need one of these to pass, it doesn't matter what's in it */
+  control_connection_t dummy;
+  /* Get results out */
+  char *answer = NULL;
+  const char *errmsg = NULL;
+
+  (void)arg;
+
+  setup_bridge_mocks();
+  MOCK(tor_mmap_file, mock_tor_mmap_file);
+  MOCK(tor_munmap_file, mock_tor_munmap_file);
+  MOCK(networkstatus_get_cache_fname,
```

Comment:
I don't think you need to mock `networkstatus_get_cache_fname()`. Just use the `get_options_mutable()` to change the options so that the function does not crash. See how other tests are doing it. e.g. `test_router_dump_router_to_string_no_bridge_distribution_method()`.

Reply:
I am trying to mock the parts of [this function](https://github.com/torproject/tor/blob/35cfe2e77679e1df5c3ae4b71112fa252acafdca/src/feature/nodelist/networkstatus.c#L245)

I don't see it using an `or_options_t` nor its callee `networkstatus_map_cached_consensus` ?